lib/Minz/Extension.php

```diff
@@ -195,4 +195,38 @@ public function registerTranslates() {
 	public function registerHook($hook_name, $hook_function) {
 		Minz_ExtensionManager::addHook($hook_name, $hook_function, $this);
 	}
+
+	/**
+	 * @return bool
+	 */
+	private function isUserConfigurationEnabled() {
+		if (!class_exists('FreshRSS_Context', false)) {
+			return false;
+		}
+		if (null === FreshRSS_Context::$user_conf) {
+			return false;
+		}
+		return true;
+	}
+
+	/**
+	 * @return array
+	 */
+	public function getUserConfiguration() {
+		if (!$this->isUserConfigurationEnabled()) {
+			return [];
+		}
+		if (!FreshRSS_Context::$user_conf->hasParam($this->getName())) {
+			return [];
+		}
+		return FreshRSS_Context::$user_conf->{$this->getName()};
+	}
+
+	public function setUserConfiguration(array $configuration) {
+		if (!$this->isUserConfigurationEnabled()) {
+			return;
+		}
+		FreshRSS_Context::$user_conf->{$this->getName()} = $configuration;
```

Comment:
Should it maybe be one lever further down, into a sub-section `->extensions` ?

Reply:
It could be. I found that it was easier this way because there is no need to manipulate the extension content. By that I mean that I cannot to something like:
```php
FreshRSS_Context::$user_conf->extension->{$this->getName()} = $configuration;
```
nor
```php
FreshRSS_Context::$user_conf->extension[{$this->getName()}] = $configuration;
```

I need to get the content of the extension first do the modifications then save.

Reply:
```php
FreshRSS_Context::$user_conf->extensions[$this->getName()] = ...;
```
should work

Reply:
I haven't try this one. I'll keep you posted.

Reply:
It does not work :/

Reply:
To make it work, I need to do something like:
```php
		$extensions = FreshRSS_Context::$user_conf->extensions;
		$extensions[$this->getName()] = $configuration;
		FreshRSS_Context::$user_conf->extensions = $extensions;
```

Reply:
For the get method, it works directly though. I will need to do something similar for the remove method. Do you think this is better?

Reply:
Arg, this is because of the indirection caused by `Minz_Configuration`. Maybe making `$user_conf->extensions` return an https://php.net/class.arrayaccess (object looking like an array) instead of a copy-on-write array (as now) could work

Reply:
@Alkarex I don't mind adding the extensions layer. I think it is a better way to organize things even if the code is less elegant. I made a first draft real quick to validate the idea. Now I can refine it :)